File: examples/key/encrypt_aes/encrypt_aes.py
from bloock.entity.encryption.encrypter import Encrypter
from bloock.entity.key.key_type import KeyType
from bloock.client.encryption import EncryptionClient
from bloock.client.record import RecordClient
from bloock.client.key import KeyClient

if __name__ == "__main__":
    payload = "This will be encrypted"
    record_client = RecordClient()
    encryption_client = EncryptionClient()
    key_client = KeyClient()

    print(f'The following payload will be encrypted: "{payload}"')

    key = key_client.new_local_key(KeyType.Aes256)

    # To encrypt a record during the building process
    encrypted_record = (
        record_client.from_string(payload)
        .with_encrypter(Encrypter(key))
        .build()
    )
    print("Encryption successful")
    print(f"Encrypted payload: {encrypted_record.retrieve().decode()}")

    # To encrypt a record independently
    record = record_client.from_string(payload).build()
    encrypted_record = encryption_client.encrypt(record, Encrypter(key))

    print("Encryption successful")
    print(f"Encrypted payload: {encrypted_record.retrieve().decode()}")

File: examples/key/encrypt_aes/EncryptAes.java
import com.bloock.sdk.client.EncryptionClient;
import com.bloock.sdk.client.KeyClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.encryption.Encrypter;
import com.bloock.sdk.entity.key.KeyType;
import com.bloock.sdk.entity.key.LocalKey;
import com.bloock.sdk.entity.record.Record;

import java.nio.charset.StandardCharsets;

public class EncryptAes {
  public static void main(String[] args) throws Exception {
    String payload = "This will be encrypted";

    System.out.println("The following payload will be encrypted: " + payload);
    RecordClient recordClient = new RecordClient();
    KeyClient keyClient = new KeyClient();

    LocalKey localKey = keyClient.newLocalKey(KeyType.Aes256);

    // To encrypt a record during the building process
    Record encryptedRecord = recordClient.fromString(payload)
        .withEncrypter(new Encrypter(localKey))
        .build();

    // To encrypt a record independently
    EncryptionClient encryptionClient = new EncryptionClient();
    Record record = recordClient.fromString(payload).build();
    encryptedRecord = encryptionClient.encrypt(record, new Encrypter(localKey));

    System.out.println("Encryption successful");
    System.out.println("Encrypted payload: " + new String(encryptedRecord.retrieve(), StandardCharsets.UTF_8));
  }
}